tools/docker/static-analysis/cppcheck.sh

```diff
@@ -1,22 +1,14 @@
 #!/bin/bash
 
-scriptdir="$(cd "${0%/*}"; pwd)"
+scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
 rootdir="${scriptdir%/*/*/*}"
 
+# shellcheck source=../../../tools/functions.sh
 . "${rootdir}/tools/functions.sh"
 
 OUTPUT_FILE="$rootdir/cppcheck_results.txt"
-INCLUDES="\
-    -I$rootdir/framework/platform/bpl/include \
-    -I$rootdir/framework/transport/ieee1905_transport/include \
-    -I$rootdir/framework/common/include \
-    -I$rootdir/framework/tlvf/src/include \
-    -I$rootdir/framework/tlvf/AutoGenerated/include \
-    -I$rootdir/common/beerocks/btl/include \
-    -I$rootdir/common/beerocks/tlvf/src/include \
-    -I$rootdir/common/beerocks/tlvf/AutoGenerated/include \
-    -I$rootdir/common/beerocks/bcl/include \
-    -I$rootdir/common/beerocks/bwl/include"
+INCLUDES=()
+while IFS='' read -r line; do INCLUDES+=("-I$line"); done < <(find -L "$rootdir/common" "$rootdir/framework" -type d -name include)
```

Comment:
Nice! We're less likely to forget to add new includes over time (or to remove old ones)...